controllers/resources/resourcestatus.go

```diff
@@ -0,0 +1,265 @@
+		Kind:       "Deployment",
+		Status:     NotReady,
+	}
+
+	deployment := &appsv1.Deployment{}
+	err := k8sClient.Get(ctx, namespacedName, deployment)
+
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			reqLogger.Error(err, "Error reading deployment for status update")
+		}
+		return
+	}
+
+	status.APIVersion = deployment.APIVersion
+
+	for _, condition := range deployment.Status.Conditions {
+		if condition.Type == appsv1.DeploymentAvailable && condition.Status == corev1.ConditionTrue {
+			status.Status = Ready
+			return
+		}
+	}
+	return
+}
+
+func getRouteStatus(ctx context.Context, k8sClient client.Client, namespacedName types.NamespacedName) (status v1alpha1.ManagedResourceStatus) {
+	reqLogger := log.WithValues("func", "getRouteStatus", "namespacedName", namespacedName)
+
+	status = v1alpha1.ManagedResourceStatus{
+		ObjectName: namespacedName.Name,
+		APIVersion: Unknown,
+		Namespace:  namespacedName.Namespace,
+		Kind:       "Route",
+		Status:     NotReady,
+	}
+	route := &routev1.Route{}
+	err := k8sClient.Get(ctx, namespacedName, route)
+
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			reqLogger.Error(err, "Error reading route for status update")
+		}
+		return
+	}
+
+	status.APIVersion = route.APIVersion
+
+	for _, routeIngress := range route.Status.Ingress {
+		for _, condition := range routeIngress.Conditions {
+			if condition.Type == routev1.RouteAdmitted && condition.Status != corev1.ConditionTrue {
+				return
+			}
+		}
+	}
+	status.Status = Ready
+	return
+}
+
+type statusRetrievalFunc func(context.Context, client.Client, []string, string) []v1alpha1.ManagedResourceStatus
+
+func getAllServiceStatus(ctx context.Context, k8sClient client.Client, names []string, namespace string) (statuses []v1alpha1.ManagedResourceStatus) {
+	reqLogger := log.WithValues("func", "getAllServiceStatus", "namespace", namespace)
+	for _, name := range names {
+		nsn := types.NamespacedName{Name: name, Namespace: namespace}
+		statuses = append(statuses, getServiceStatus(ctx, k8sClient, nsn))
+	}
+	reqLogger.Info("New statuses", "statuses", statuses)
+	return
+}
+
+func getAllDeploymentStatus(ctx context.Context, k8sClient client.Client, names []string, namespace string) (statuses []v1alpha1.ManagedResourceStatus) {
+	reqLogger := log.WithValues("func", "getAllDeploymentStatus", "namespace", namespace)
+	for _, name := range names {
+		nsn := types.NamespacedName{Name: name, Namespace: namespace}
+		statuses = append(statuses, getDeploymentStatus(ctx, k8sClient, nsn))
+	}
+	reqLogger.Info("New statuses", "statuses", statuses)
+	return
+}
+
+func getAllRouteStatus(ctx context.Context, k8sClient client.Client, names []string, namespace string) (statuses []v1alpha1.ManagedResourceStatus) {
+	reqLogger := log.WithValues("func", "getAllRouteStatus", "namespace", namespace)
+	for _, name := range names {
+		r := &routev1.Route{}
+		nsn := types.NamespacedName{Name: name, Namespace: namespace}
+		_ = k8sClient.Get(ctx, nsn, r)
```

Comment:
You don't need this call anymore since it's happening in getRouteStatus now.

Reply:
fixed